xarray/core/dataset.py

```diff
@@ -359,6 +359,39 @@ def _assert_empty(args: tuple, msg: str = "%s") -> None:
         raise ValueError(msg % args)
 
 
+def _selkeys(dict_, keys):
```

Comment:
I see that you only moved this helper function out, but I would take the occasion to move the logic back into `_maybe_chunk`.